struct.go

```diff
@@ -329,15 +329,40 @@ func (k K) String() string {
 	}
 }
 
-// Message is malformated or invalid
-var ErrBadMsg = errors.New("Bad Message")
-
 // Message header is invalid
 var ErrBadHeader = errors.New("Bad header")
 
 // Cannot process sync requests
 var ErrSyncRequest = errors.New("nosyncrequest")
 
+// NewQError initializes a new QError with a message.
+func NewUnsupportedTypeError(msg string) error {
+	return &UnsupportedTypeError{msg}
+}
+
+// UnsupportedTypeError is an error returned by Fn() calls. It can be used to distinguish these errors from generic errors
```

Comment:
(minor) `Fn()` is a concept in Appian gateways. I think it will be confusing mentioning it here. I think it will be better to say something along the lines of

`error returned when encoding or decoding a type that is not supported by kdb`